src/vs/base/parts/tree/browser/tree.ts

```diff
@@ -500,7 +502,8 @@ export interface IDragAndDrop {
 	/**
 	 * Sent when the drag operation is starting.
 	 */
-	onDragStart(tree: ITree, data: IDragAndDropData, originalEvent: Mouse.DragMouseEvent): void;
+	// {{SQL CARBON EDIT}}
+	onDragStart(tree: ITree, data: IDragAndDropData, originalEvent: Mouse.DragMouseEvent, element: any, item: ViewItem): void;
```

Comment:
You don't need to change the interface here. You can get the `element` from `data`, as it's always going to be the first element in the array if you're support drag and drop of only a single element at once. You can also get rid of `item` since you can retrieve `item.uri` as `data.elements[0].id`

Reply:
![Screen Shot 2020-06-12 at 1 27 26 PM](https://user-images.githubusercontent.com/18150417/84530031-75df2000-acb0-11ea-8eec-52c61d5c8deb.png)
When I try to use `data.elements[0].id` it gives me an error saying `Property 'elements' does not exist on type 'IDragAndDropData'.` even though it seems like the code works when I run through with the debugger

Reply:
From your screenshot, I can see that `data`'s type is `ElementsDragAndDropData`, which exists in `src/vs/base/browser/ui/list/listView.ts`, which just implements `IDragAndDropData`. You could either change the function parameter type to be `ElementsDragAndDropData` or you could do something like `(data as ElementsDragAndDropData).elements`